ckanext/s3filestore/plugin.py

```diff
@@ -60,12 +60,14 @@ def get_uploader(self, upload_to, old_filename=None):
     def before_map(self, map):
         with SubMapper(map, controller='ckanext.s3filestore.controller:S3Controller') as m:
             # Override the resource download links
-            m.connect('resource_download',
-                      '/dataset/{id}/resource/{resource_id}/download',
-                      action='resource_download')
-            m.connect('resource_download',
-                      '/dataset/{id}/resource/{resource_id}/download/{filename}',
-                      action='resource_download')
+            # if plugins.toolkit.check_ckan_version(max_version='2.8.2') & \
+            if hasattr(DefaultResourceUpload, "download") == False :
```

Comment:
How is the download URL routed when the `download` function *does* exist?

Reply:
ckan/controllers/package.py 
is changed to
return upload.download(rsc['id'], filename)
from
resource download bits.